Index: mailcommon/src/filter/autotests/CMakeLists.txt
===================================================================
--- mailcommon.orig/src/filter/autotests/CMakeLists.txt
+++ mailcommon/src/filter/autotests/CMakeLists.txt
@@ -245,9 +245,3 @@ add_mailcommon_filter_test(filteractionm
     filteractionmissingtagdialogtest.cpp
     ../dialog/filteractionmissingtagdialog.cpp
     )
-
-add_mailcommon_filter_test(filteractionmissingidentitydialogtest
-    filteractionmissingidentitydialogtest.cpp
-    ../dialog/filteractionmissingidentitydialog.cpp
-    ../../../autotests/dummykernel.cpp
-    )
